src/sql/azdata.proposed.d.ts

```diff
@@ -4416,7 +4416,7 @@ declare module 'azdata' {
 			/**
 			 * Kicks off execution of all code cells. Thenable will resolve only when full execution of all cells is completed.
 			 */
-			runAllCells(): Thenable<boolean>;
+			runAllCells(startId?: string, endId?: string): Thenable<boolean>;
```

Comment:
So - near as I can tell, we don't expose a cell ID. We only have URI, and contents. Given this I would recommend either not plumbing through, OR adding in ID, OR using URI instead. What do you think? 

```ts
		export interface NotebookCell {
			contents: ICellContents;
			uri?: vscode.Uri;
		}
```

Reply:
Let's go with NotebookCell, since that's what runCell uses